lib/graphql/internal_representation/rewrite.rb

```diff
@@ -45,6 +45,12 @@ def validate(context)
         visitor[Nodes::Field].enter << -> (ast_node, prev_ast_node) {
           parent_node = @nodes.last
           node_name = ast_node.alias || ast_node.name
+          
+          return :_skip if ast_node.directives.any? do |directive|
+            directive.name == 'skip' && directive.arguments.any? {|arg| arg.value == true } ||
+            directive.name == 'include' && directive.arguments.any? {|arg| arg.value == false }
+          end
```

Comment:
Just to make sure I understand correctly, this is here so that, when we _know_ that this field won't be included because of `@skip`/`@include`, we can avoid building an IRep tree, since we _know_ we won't be executing it.  Is that right? 

If this optimization is separate from the document validation issue, do you mind extracting it to a separate PR? In my opinion this kind of optimization deserves a serious look in its own right!